tests/storages_tests/rdb_tests/test_storage.py

```diff
@@ -177,6 +184,72 @@ def test_upgrade() -> None:
     assert old_version == new_version
 
 
+@pytest.mark.parametrize("optuna_version", ["0.9.0", "1.2.0", "1.3.0", "2.4.0", "2.6.0"])
+def test_upgrade(optuna_version: str) -> None:
+    src_db_file = os.path.join(
+        os.path.dirname(__file__), "test_upgrade_assets", f"{optuna_version}.db"
+    )
+    with tempfile.TemporaryDirectory() as workdir:
+        shutil.copyfile(src_db_file, f"{workdir}/sqlite.db")
+        storage_url = f"sqlite:///{workdir}/sqlite.db"
+
+        storage = RDBStorage(storage_url, skip_compatibility_check=True)
+        assert storage.get_current_version() == f"v{optuna_version}.a"
+        head_version = storage.get_head_version()
+        storage.upgrade()
+        assert head_version == storage.get_current_version()
+
+        # Create a new study.
+        study = create_study(storage=storage)
+        assert len(study.trials) == 0
+        study.optimize(objective_test_upgrade, n_trials=1)
+        assert len(study.trials) == 1
+
+        # Check empty study.
+        study = load_study(storage=storage, study_name="single_empty")
+        assert len(study.trials) == 0
+        study.optimize(objective_test_upgrade, n_trials=1)
+        assert len(study.trials) == 1
+
+        # Resume single objective optimization.
+        study = load_study(storage=storage, study_name="single")
+        assert len(study.trials) == 1
+        study.optimize(objective_test_upgrade, n_trials=1)
+        assert len(study.trials) == 2
+        for trial in study.trials:
+            assert trial.system_attrs["a"] == 0
+            assert trial.user_attrs["b"] == 1
+            assert trial.intermediate_values[0] == 0.5
```

Comment:
I'm wondering if we should check value(s), params, and such properties as well (covering all RDB tables).

Reply:
Thank you for your suggestion. I added the assertions of params and values. At first, I used `enqueue_trial` to fix those values, but I couldn't use it in optuna==0.7.0, and I gave up.

Reply:
I think that's great, thanks!